CatalogReaderExample/NuGet.Protocol.Catalog/CatalogProcessor.cs

```diff
@@ -0,0 +1,214 @@
+using Microsoft.Extensions.Logging;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol.Catalog
+{
+    public class CatalogProcessor
+    {
+        private const string CatalogResourceType = "Catalog/3.0.0";
+        private readonly ICatalogLeafProcessor _leafProcessor;
+        private readonly ICatalogClient _client;
+        private readonly ICursor _cursor;
+        private readonly ILogger<CatalogProcessor> _logger;
+        private readonly CatalogProcessorSettings _settings;
+
+        public CatalogProcessor(
+            ICursor cursor,
+            ICatalogClient client,
+            ICatalogLeafProcessor leafProcessor,
+            CatalogProcessorSettings settings,
+            ILogger<CatalogProcessor> logger)
+        {
+            _leafProcessor = leafProcessor ?? throw new ArgumentNullException(nameof(leafProcessor));
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _cursor = cursor ?? throw new ArgumentNullException(nameof(cursor));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            if (settings.ServiceIndexUrl == null)
+            {
+                throw new ArgumentException(
+                    $"The {nameof(CatalogProcessorSettings.ServiceIndexUrl)} property of the " +
+                    $"{nameof(CatalogProcessorSettings)} must not be null.");
+            }
+
+            // Clone the settings to avoid mutability issues.
+            _settings = new CatalogProcessorSettings
+            {
+                ServiceIndexUrl = settings.ServiceIndexUrl,
+                DefaultMinCommitTimestamp = settings.DefaultMinCommitTimestamp,
+                MinCommitTimestamp = settings.MinCommitTimestamp,
+                MaxCommitTimestamp = settings.MaxCommitTimestamp,
+                ExcludeRedundantLeaves = settings.ExcludeRedundantLeaves,
+            };
+        }
+
+        public async Task ProcessAsync()
+        {
+            var catalogIndexUrl = await GetCatalogIndexUrlAsync();
+
+            var minCommitTimestamp = await GetMinCommitTimestamp();
+            _logger.LogInformation(
+                "Using time bounds {min:O} (exclusive) to {max:O} (inclusive).",
+                minCommitTimestamp,
+                _settings.MaxCommitTimestamp);
+
+            await ProcessIndexAsync(catalogIndexUrl, minCommitTimestamp);
+        }
+
+        private async Task ProcessIndexAsync(string catalogIndexUrl, DateTimeOffset minCommitTimestamp)
+        {
+            var index = await _client.GetIndexAsync(catalogIndexUrl);
+
+            var pageItems = index.GetPagesInBounds(
+                minCommitTimestamp,
+                _settings.MaxCommitTimestamp);
+            _logger.LogInformation(
+                "{pages} pages were in the time bounds, out of {totalPages}.",
```

Comment:
> "{pages} pages were in the time bounds, out of {totalPages}.", [](start = 15, length = 63)

I thought this format works only with Serilog..

Reply:
`Microsoft.Extensions.Logging` supports this as well.